components/motor/ulnstepper/28byj-48.go

```diff
@@ -0,0 +1,478 @@
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/operation"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+var (
+	model                = resource.NewDefaultModel("28byj48")
+	minStepDelay         = 0.002                  // minimum sleep time between steps
+	minDelayBetweenTicks = 100 * time.Microsecond // minimum sleep time between each ticks
+)
+
+// stepSequence contains high and low signal for uln2003 pins.
+// Treversing through stepSequence once is one step.
+var stepSequence = [8][4]bool{
+	{true, false, false, true},
+	{true, false, false, false},
+	{true, true, false, false},
+	{false, true, false, false},
+	{false, true, true, false},
+	{false, false, true, false},
+	{false, false, true, true},
+	{false, false, false, true},
+}
+
+// PinConfig defines the mapping of where motor are wired.
+type PinConfig struct {
+	In1 string `json:"In1"`
+	In2 string `json:"In2"`
+	In3 string `json:"In3"`
+	In4 string `json:"In4"`
+}
+
+// Config describes the configuration of a motor.
+type Config struct {
+	Pins              PinConfig `json:"pins"`
+	BoardName         string    `json:"board"`
+	RotationPerMinute float64   `json:"rotation_per_minute"`
+	TicksPerRotation  int       `json:"ticks_per_rotation"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (config *Config) Validate(path string) ([]string, error) {
+	var deps []string
+	if config.BoardName == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "board")
+	}
+	deps = append(deps, config.BoardName)
+	return deps, nil
+}
+
+func init() {
+	_motor := registry.Component{
+		Constructor: func(ctx context.Context, deps registry.Dependencies, config config.Component, logger golog.Logger) (interface{}, error) {
+			actualBoard, motorConfig, err := getBoardFromRobotConfig(deps, config)
+			if err != nil {
+				return nil, err
+			}
+
+			return newULN(ctx, actualBoard, *motorConfig, config.Name, logger)
+		},
+	}
+	registry.RegisterComponent(motor.Subtype, model, _motor)
+	config.RegisterComponentAttributeMapConverter(
+		motor.Subtype,
+		model,
+		func(attributes config.AttributeMap) (interface{}, error) {
+			var conf Config
+			return config.TransformAttributeMapToStruct(&conf, attributes)
+		},
+		&Config{},
+	)
+}
+
+func getBoardFromRobotConfig(deps registry.Dependencies, config config.Component) (board.Board, *Config, error) {
+	motorConfig, ok := config.ConvertedAttributes.(*Config)
+	if !ok {
+		return nil, nil, rdkutils.NewUnexpectedTypeError(motorConfig, config.ConvertedAttributes)
+	}
+	if motorConfig.BoardName == "" {
+		return nil, nil, errors.New("expected board name in config for motor")
+	}
+	b, err := board.FromDependencies(deps, motorConfig.BoardName)
+	if err != nil {
+		return nil, nil, err
+	}
+	return b, motorConfig, nil
+}
+
+func newULN(ctx context.Context, b board.Board, mc Config, name string, logger golog.Logger) (motor.Motor, error) {
+	if mc.TicksPerRotation == 0 {
```

Comment:
I am thinking of making TicksPerRotation a constant in the driver since 4096 ticks completes one revolution. Anything less than this will cause the motor to not complete one revolution. @gvaradarajan agrees with me. I want to get your and @randhid 's opinion on it.